src/pages/ProductListingPage.js

```diff
@@ -1,17 +1,54 @@
 import './ProductListingPage.css';
-import { Fragment } from 'react';
+import { Fragment, useEffect } from 'react';
 import { Footer, Header, Filters, Listing } from '../components';
-import products from '../data/inventory.json';
+import { useParams } from 'react-router-dom';
+import { useState } from 'react';
 
 const ProductListingPage = props => {
+  const [products, setProducts] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const params = useParams();
+
+  const category = params.category;
+  const collection = params.collection;
```

Comment:
you can replace this like  suppose eg : const {category, collection} =useParams();